tests/e2e/vmgroupmisc_test.go

```diff
@@ -0,0 +1,261 @@
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, false, Commentf("Failed to delete the vmgroup [%s] .", vmGroupName))
+
+	// Removing the DB at the end of suite
+	admincli.ConfigRemove(s.config.EsxHost)
+
+	// Verifying DB successfully removed
+	c.Assert(admincli.GetDBmode(s.config.EsxHost), Equals, admincliconst.DBNotConfigured, Commentf("Failed to remove the DB mode on ESX -  .", s.config.EsxHost))
+
+}
+
+var _ = Suite(&vgBasicSuite{})
+
+// Verify vmgroup for orphaned volumes is specified as N/A
+// 1. Create a vmgroup and add vm to it.
+// 2. Verify vmgroup was successfully created.
+// 3. Create a volume from vm-2.
+// 4. Verify volumes vmgroup from esx.
+// 5. Remove vm-2 from vmgroup.
+// 6. Verify volume is not visible from vm-2.
+// 7. Delete the tenant.
+// 8. Again verify volumes vmgroup is listed as N/A.
+func (s *vgBasicSuite) TestVGNameForOrphanedVolumes(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume from vmgroup's vm
+	out, err := dockercli.CreateVolume(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Check if volume was successfully created
+	isAvailable := verification.CheckVolumeAvailability(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", s.volumeNames[0]))
+
+	// Verify if volume exists in vmgroup
+	isVolInVmgroup := admincli.IsVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], vmGroupName)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Volume [%s] does not belong to vmgroup [%s]", s.volumeNames[0], vmGroupName))
+
+	admincli.RemoveVMFromVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[0])
+
+	// Check if volume was not visible from vm since it does not belong to vmgroup
+	isAvailable = verification.CheckVolumeAvailability(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(isAvailable, Equals, false, Commentf("Unexpected Behavior: Volume %s belonging to "+
+		"vmgroup [%s] is visible from host [%s] which does not belong to the same vmgroup", s.volumeNames[0], vmGroupName, s.config.DockerHosts[0]))
+
+	// Now delete the vmgroup
+	admincli.DeleteVMgroup(s.config.EsxHost, vmGroupName)
+
+	// Verify vmgroup does not exist
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, false, Commentf("Failed to delete the vmgroup [%s] .", vmGroupName))
+
+	// Verify vmgroup for volume is N/A
+	isVolInVmgroup = admincli.IsVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], admincliconst.NotApplicable)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Unexpected Behavior: Vmgroup [%s] for volume [%s] is not N/A. ", s.volumeNames[0], vmGroupName))
+
+	// Delete the orphaned volume which does not belong to any vmgroup
+	admincli.DeleteOrphanedVolume(s.config.EsxHost, s.volumeNames[0], vmGroupName, s.config.Datastores[0])
```

Comment:
Yes, we are not able to delete volumes for which there are no vmgroups - N/A. I will file an issue for it. Wrote this util to not pollute our test setup else we will have left-over volumes.